### DISCUSSION TITLE
enh: Web search should be used judiciously, as an available tool

### DISCUSSION THREAD

**Discussion by lee-b** [open-webui/open-webui#4851] [General]
When "Web search" is selected from the "+" icon in the message box, it seems to be done every single time, and can even reveal the user's discussion directly (which defeats the purpose of an offline model).  For example, when responding to the AI with "that's not the case", with the web search option still selected, the Open WebUI will send that response to the web verbatim as a search.

Instead, I think the websearch should be provided as an optional tool to the AI, for it to use if it needs it, or even to use multiple times if the message requires getting an overview and also some details, for example.

There's probably a workaround here, along the lines of selecting the websearch only when you need it, and/or configuring tools in pipelines and/or the Workspace instead, but I think the Web Search option in the message box ("+" icon) should do the right thing here, enabling a tool in the workspace automatically, or whatever.

**Comment by lee-b** [comment]
As an aside, I notice that web searches are extremely verbatim and based on the last message, even when desired.  Do they take the current chat context into account _at all_?

**Comment by dnafication** [comment]
It does seem to generate a search query for some, I wonder if the text length plays any role in this.

Example below generates a shorter search query:
![image](https://github.com/user-attachments/assets/795498ff-35cd-4015-92b2-e0174c4760a6)

However, questions like below is forwarded as is:
![image](https://github.com/user-attachments/assets/f847d94f-7d17-4e3b-96a6-6776a4957bee)

**Comment by justinh-rahb** [comment]
@dnafication 
<img width="662" alt="Screenshot 2024-08-06 at 1 19 01 AM" src="https://github.com/user-attachments/assets/644bb801-2cef-4d42-9aab-d8b6238a4432">

**Comment by tjbck** [comment]
Some local model (notably lower-end models) are too unreliable for this task, turning this to discussion this but if anyone manages to make this feature work reliably, feel free to make a PR.

**Reply by tjbck** [reply]
Updated with latest dev!
